metron-platform/metron-elasticsearch/src/test/java/org/apache/metron/elasticsearch/integration/components/ElasticSearchComponent.java

```diff
@@ -194,35 +215,41 @@ public Client getClient() {
     return client;
   }
 
-  public BulkResponse add(String indexName, String sensorType, String... docs) throws IOException {
+  public void add(UpdateDao updateDao, String indexName, String sensorType, String... docs)
+          throws IOException, ParseException {
     List<String> d = new ArrayList<>();
     Collections.addAll(d, docs);
-    return add(indexName, sensorType, d);
+    add(updateDao, indexName, sensorType, d);
   }
 
-  public BulkResponse add(String indexName, String sensorType, Iterable<String> docs)
-      throws IOException {
-    BulkRequestBuilder bulkRequest = getClient().prepareBulk();
-    for (String doc : docs) {
-      IndexRequestBuilder indexRequestBuilder = getClient()
-          .prepareIndex(indexName, sensorType + "_doc");
-
-      indexRequestBuilder = indexRequestBuilder.setSource(doc);
-      Map<String, Object> esDoc = JSONUtils.INSTANCE
-          .load(doc, JSONUtils.MAP_SUPPLIER);
-      indexRequestBuilder.setId((String) esDoc.get(Constants.GUID));
-      Object ts = esDoc.get("timestamp");
-      if (ts != null) {
-        indexRequestBuilder = indexRequestBuilder.setTimestamp(ts.toString());
-      }
-      bulkRequest.add(indexRequestBuilder);
-    }
+  public void add(UpdateDao updateDao, String indexName, String sensorType, Iterable<String> docs)
```

Comment:
The Elasticsearch tests use this method to load test data.  Rather than using the Transport client here, we use the `UpdateDao` to load the test data.

Reply:
Might it be better to just use IndexDao, which `extends UpdateDao, SearchDao, RetrieveLatestDao, ColumnMetadataDao`? To that end, if we're looking to route all of this through the ES component in that fashion, it might make sense to simply replace the internal `private Client client;` and instead use the new desired IndexDao for the proxied calls to ES. It could be setup at construction time of the component and remove the need to do the same thing for every test that uses the component class, unless they actually want to do something custom and pass in their own dao during init.

Reply:
> Might it be better to just use IndexDao, which extends UpdateDao, SearchDao, RetrieveLatestDao, ColumnMetadataDao? 

Why would that be better?  It needs to do updates, so it needs an `UpdateDao`.  Seemed logical to me.

Reply:
> To that end, if we're looking to route all of this through the ES component in that fashion, it might make sense to simply replace the internal private Client client; and instead use the new desired IndexDao for the proxied calls to ES.

I don't think we're ready to do all that quite yet.  There is still some legacy functionality in `ElasticSearchComponent` that uses the underlying client for the old Admin API.  See [close](https://github.com/apache/metron/blob/fcd644ca77394d48d460c460b672a23d6594f49b/metron-platform/metron-elasticsearch/src/test/java/org/apache/metron/elasticsearch/integration/components/ElasticSearchComponent.java#L292), [createIndexWithMapping](https://github.com/apache/metron/blob/fcd644ca77394d48d460c460b672a23d6594f49b/metron-platform/metron-elasticsearch/src/test/java/org/apache/metron/elasticsearch/integration/components/ElasticSearchComponent.java#L228), and [start](https://github.com/apache/metron/blob/fcd644ca77394d48d460c460b672a23d6594f49b/metron-platform/metron-elasticsearch/src/test/java/org/apache/metron/elasticsearch/integration/components/ElasticSearchComponent.java#L153).

Reply:
The IndexDao handles all of the plumbing for the init that you're duplicating in this test class.